components/function-controller/pkg/apis/serverless/v1alpha2/function_defaults.go

```diff
@@ -58,30 +58,34 @@ type DefaultingConfig struct {
 }
 
 func (fn *Function) Default(config *DefaultingConfig) {
-	fn.Spec.defaultReplicas(config)
+	fn.Spec.defaultScaling(config)
 	fn.Spec.defaultFunctionResources(config, fn)
 	fn.Spec.defaultBuildResources(config, fn)
 }
 
-func (spec *FunctionSpec) defaultReplicas(config *DefaultingConfig) {
-	if spec.Replicas != nil {
-		return
-	}
-
+func (spec *FunctionSpec) defaultScaling(config *DefaultingConfig) {
 	defaultingConfig := config.Function.Replicas
 	replicasPreset := defaultingConfig.Presets[defaultingConfig.DefaultPreset]
 
+	if spec.Replicas == nil {
+		// TODO: The presets structure and docs should be updated to reflect the new behavior.
+		spec.Replicas = &replicasPreset.Min
+	}
+
 	if spec.ScaleConfig == nil {
-		spec.ScaleConfig = &ScaleConfig{}
+		return
 	}
+
+	// spec.ScaleConfig is SET, but not fully configured, for sanity, we will default MinReplicas, we will also use it as a default spec.Replica
```

Comment:
I am preserving as much as possible from the original behavior. Since I didn't really cover validation in the proposal doc. Since there is a [dedicated issue](https://github.com/kyma-project/kyma/issues/14900). I prefer that we address it there.

Reply:
Added: https://github.com/kyma-project/kyma/issues/14900#issuecomment-1273095981